File: projects/agentic-automation-framework-python/agentic/channels/telegram.py
import requests


BASE = "https://api.telegram.org"


def _call(token: str, method: str, payload: dict):
    r = requests.post(f"{BASE}/bot{token}/{method}", json=payload, timeout=40)
    data = r.json()
    if not data.get("ok"):
        raise RuntimeError(f"Telegram {method} failed: {data}")
    return data["result"]


def send_message(token: str, chat_id: str, text: str):
    if not token or not chat_id:
        return None
    return _call(token, "sendMessage", {"chat_id": chat_id, "text": text, "disable_web_page_preview": True})


def poll_updates(token: str, offset=0, timeout=30):
    if not token:
        return []
    return _call(token, "getUpdates", {"offset": offset, "timeout": timeout})


def parse_command(update: dict):
    msg = update.get("message") or {}
    txt = msg.get("text")
    if not txt:
        return None
    return {
        "update_id": update.get("update_id"),
        "text": txt.strip(),
        "chat_id": str(msg.get("chat", {}).get("id", "")),
        "user_id": str((msg.get("from") or {}).get("id", "")),
    }

File: projects/agentic-automation-framework/src/channels/telegram.js
const BASE = "https://api.telegram.org";

async function tg(token, method, payload = {}) {
  const res = await fetch(`${BASE}/bot${token}/${method}`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload)
  });
  const data = await res.json();
  if (!data.ok) throw new Error(`Telegram ${method} failed: ${JSON.stringify(data)}`);
  return data.result;
}

export async function sendTelegramMessage({ token, chatId, text }) {
  if (!token || !chatId) return;
  return tg(token, "sendMessage", { chat_id: chatId, text, disable_web_page_preview: true });
}

export async function pollTelegramUpdates({ token, offset = 0, timeout = 30 }) {
  if (!token) return [];
  return tg(token, "getUpdates", { offset, timeout });
}

export function parseTelegramCommand(update) {
  const msg = update?.message;
  if (!msg?.text) return null;
  return {
    updateId: update.update_id,
    text: msg.text.trim(),
    chatId: String(msg.chat.id),
    userId: String(msg.from?.id || "")
  };
}
